src/tensorlake/functions_sdk/graph.py

```diff
@@ -234,6 +235,14 @@ def definition(self) -> ComputeGraphMetadata:
                     delay_multiplier=start_node.retries.delay_multiplier,
                 )
             ),
+            cache_key=(
+                CacheKeyMetadata(
+                    algorithm=start_node.cache_key["algorithm"],
+                    value=start_node.cache_key["value"],
+                )
+                if hasattr(start_node, "cache_key") and start_node.cache_key is not None
```

Comment:
iirc this `hasattr(start_node, "cache_key")` is not needed, is there a bug somewhere?

Reply:
It was needed for nodes that weren't functions; that was fixed by adding the field to `TensorlakeCompute`.